src/mlpack/methods/decision_tree/random_binary_numeric_split.hpp

```diff
@@ -0,0 +1,117 @@
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace tree {
+
+/**
+ * The RandomBinaryNumericSplit is a splitting function for decision trees that
+ * will split based on a randomly selected point between the minimum
+ * and maximum value of the numerical dimension.
+ *
+ * @tparam FitnessFunction Fitness function to use to calculate gain.
+ */
+template<typename FitnessFunction>
+class RandomBinaryNumericSplit
+{
+ public:
+  // No extra info needed for split.
+  template<typename ElemType>
+  class AuxiliarySplitInfo { };
+
+  /**
+   * Check if we can split a node.  If we can split a node in a way that
+   * improves on 'bestGain', then we return the improved gain.  Otherwise we
+   * return the value 'bestGain'.  If a split is made, then classProbabilities
+   * and aux may be modified.
+   *
+   * @param bestGain Best gain seen so far (we'll only split if we find gain
+   *      better than this).
+   * @param data The dimension of data points to check for a split in.
+   * @param labels Labels for each point.
+   * @param numClasses Number of classes in the dataset.
+   * @param weights Weights associated with labels.
+   * @param minimumLeafSize Minimum number of points in a leaf node for
+   *      splitting.
+   * @param minimumGainSplit Minimum gain split.
+   * @param classProbabilities Class probabilities vector, which may be filled
+   *      with split information a successful split.
+   * @param aux Auxiliary split information, which may be modified on a
+   *      successful split.
+   */
+  template<bool UseWeights, typename VecType, typename WeightVecType>
+  static double SplitIfBetter(
+      const double bestGain,
+      const VecType& data,
+      const arma::Row<size_t>& labels,
+      const size_t numClasses,
+      const WeightVecType& weights,
+      const size_t minimumLeafSize,
+      const double minimumGainSplit,
+      arma::Col<typename VecType::elem_type>& classProbabilities,
+      AuxiliarySplitInfo<typename VecType::elem_type>& aux);
+
+  /**
+   * Returns 2, since the binary split always has two children.
+   */
+  template<typename ElemType>
+  static size_t NumChildren(const arma::Col<ElemType>& /* classProbabilities */,
+                            const AuxiliarySplitInfo<ElemType>& /* aux */)
+  {
+    return 2;
+  }
+
+  /**
+   * Given a point, calculate which child it should go to (left or right).
+   *
+   * @param point Point to calculate direction of.
+   * @param classProbabilities Auxiliary information for the split.
+   * @param * (aux) Auxiliary information for the split (Unused).
+   */
+  template<typename ElemType>
+  static size_t CalculateDirection(
+      const ElemType& point,
+      const arma::Col<ElemType>& classProbabilities,
+      const AuxiliarySplitInfo<ElemType>& /* aux */);
+};
+
+} // namespace tree
+} // namespace mlpack
+
+// Include implementation.
+#include "random_binary_numeric_split_impl.hpp"
+
+#endif
```

Comment:
This is a minor pedantic point, but can you make sure files end with a newline please? :)